scripts/sanity_chk/sanitylib.py

```diff
@@ -891,23 +891,31 @@ def handle(self):
                 #sometimes QEMU can't handle SIGTERM signal correctly
                 #in that case kill -9 QEMU process directly and leave
                 #sanitycheck judge testing result by console output
+                state, _ = self.get_state()
                 if os.path.exists(self.pid_fn):
                     qemu_pid = int(open(self.pid_fn).read())
                     try:
                         os.kill(qemu_pid, signal.SIGKILL)
                     except ProcessLookupError:
                         pass
                     proc.wait()
-                    if harness.state == "passed":
+                    if state == "passed":
                         self.returncode = 0
                     else:
-                        self.returncode = proc.returncode
+                        self.returncode = 1
                 else:
                     proc.terminate()
                     proc.kill()
-                    self.returncode = proc.returncode
+                    if state == "passed":
+                        self.returncode = proc.returncode
+                    else:
+                        self.returncode = 1
             else:
-                self.returncode = proc.returncode
+                state, _ = self.get_state()
+                if state == "passed":
+                    self.returncode = proc.returncode
+                else:
+                    self.returncode = 1
```

Comment:
After staring at the increased cyclomatic complexity for a couple minutes I think I found a simple way to reduce it:

```python
logger.debug("Spawning QEMUHandler Thread for %s" % self.name)
self.returncode = 1 # <===
try:
   proc.wait(self.timeout)
except:
```

Then you can drop all the `else` clauses which should make this much easier to read.